### DISCUSSION TITLE
NU1902 and 1903 Warnings as Error buiding ApsimX with Visual Studio 17.12.2

### DISCUSSION THREAD

**Discussion by pjinnes** [APSIMInitiative/ApsimX#9478] [Q&A] [answered]
I updated my community Visual Studio to latest rev. 17.12.2 and now can't compile ApsimX due to multiple warnings as errors such as:
Models\Models.csproj : error NU1902: Warning As Error: Package 'SixLabors.ImageSharp' 1.0.4 has a known moderate severity vulnerability
.
Models\Models.csproj : error NU1903: Warning As Error: Package 'System.Text.Json' 6.0.3 has a known high severity vulnerability,
3>Done building project "Models.csproj" -- FAILED.

I can't find a way of going back to an earlier VS or telling it to ignore the warnings

**Comment by lie112** [comment]
The errors will show up beside the packages in your solution explorer (Models\dependencies\packages). The you can add NU1092 and NU1093 as "supress warnings" for these two issues in their properties section (right click\properties). They will then be ignored during your build process and you can work until these are addressed with a full fix.

**Reply by pjinnes** [reply]
Thanks for that Adam, it turned some of my errors into warnings but I had others e.g. in Documentation that I couldn't get rid of. However, I managed to roll back my VS by running the setup file again and selecting a rollback to previous version option, so can compile again for the time being..

**Comment by par456** [comment, marked as answer]
Looks like it's time for us to update from .net 6 to  .net 8 as our packages are going out of support on .net 6.

PR #9501 will resolve this for now, but we'll have to do a full update on packages when we go across.
